### PR TITLE
Fix GridHelper colorCenterLine

### PR DESCRIPTION
**Pull request by sreith1**
Loop counter i may not equal 0 due to floating point precision e. g.
`new THREE.GridHelper( 1, 0.1, 0x0000ff, 0xff0000 );`

### CODE REVIEW

File: src/extras/helpers/GridHelper.js [hunk 1]
```diff
@@ -15,7 +15,7 @@ THREE.GridHelper = function ( size, step, color1, color2 ) {
 		vertices.push( - size, 0, i, size, 0, i );
 		vertices.push( i, 0, - size, i, 0, size );
 
-		var color = i === 0 ? color1 : color2;
+		var color = ( Math.abs( i ) < Math.pow( 2, - 50 ) ) ? color1 : color2;
```
Comment (WestLangley) [inline src/extras/helpers/GridHelper.js]: I would not expect that to work in general. For example:

    new THREE.GridHelper( 10.1, 1.01, 0x0000ff, 0xff0000 )

**Comment by WestLangley** [comment]
It would be best, IMHO, if `GridHelper` had a similar API as `PlaneGeometry`.

(...and perhaps oriented to face the postive z-axis so that `lookAt()` will work properly.)

**Comment by mrdoob** [comment]
> It would be better, IMHO, if `GridHelper` had an API similar to `PlaneGeometry`.

>     THREE.GridHelper( width, height, widthSegments, heightSegments, color1, color2 )

> or, more simply,

>     THREE.GridHelper( width, segments, color1, color2 )

I agree. I've wanted to change that for a while but the backwards compatibility issue always worries me. I guess we'll have to "break" stuff on this one.

> (...and perhaps oriented to face the postive z-axis so that `lookAt()` will work properly.)

Not sure about this one...?

**Comment by WestLangley** [comment]
>(...and perhaps oriented to face the postive z-axis so that lookAt() will work properly.)
>>Not sure about this one...?

OK. That is not a big deal.

**Comment by mrdoob** [comment]
Changed.

@WestLangley Maybe `divisions` is more clear than `segments`?

**Comment by WestLangley** [comment]
>@WestLangley Maybe divisions is more clear than segments?

Agreed.

I would also change `size` to `width`. If you don't want to do that, then I think it would be a good idea to make it clear in the docs that `size` is the half-width.

**Comment by mrdoob** [comment]
Yeah, `size` is weird. But enough breakage for now 😁
We should definitely document it though.

**Comment by mrdoob** [comment]
Closing this one as logic for figuring out the center has changed. Thanks @sreith1!

**Comment by sreith1** [comment]
That's of course a much better solution. Thank you!

**Pull request closed without merging**